### ISSUE TITLE
PhantomJS crashes when the loaded JS code contains the string </script>

### ISSUE THREAD

**Issue by axos88** [scala-js/scala-js#2279]
Starting from https://github.com/sjrd/scala-js-example-app, upgrading to latest scalajs (0.6.7) and adding this to the build.sbt:
```
jsDependencies ++= Seq(
    RuntimeDOM
    ,"org.webjars.bower" % "angular" % "1.4.9" / "angular.js"
  )

scalaJSUseRhino in Global := false
```

Causes sbt test to fail: 

```
akovanm0:scala-js-example-app avandra$ sbt test
[info] Loading project definition from projects/scala-js-example-app/project
[info] Set current project to Example (in build file:projects/scala-js-example-app/)
org.scalajs.jsenv.ExternalJSEnv$NonZeroExitException: PhantomJS exited with code 2
	at org.scalajs.jsenv.ExternalJSEnv$AbstractExtRunner.waitForVM(ExternalJSEnv.scala:110)
	at org.scalajs.jsenv.ExternalJSEnv$ExtRunner.run(ExternalJSEnv.scala:159)
	at org.scalajs.sbtplugin.FrameworkDetector.detect(FrameworkDetector.scala:66)
	at org.scalajs.sbtplugin.ScalaJSPluginInternal$$anonfun$56.apply(ScalaJSPluginInternal.scala:705)
	at org.scalajs.sbtplugin.ScalaJSPluginInternal$$anonfun$56.apply(ScalaJSPluginInternal.scala:688)
	at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
	at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:40)
	at sbt.std.Transform$$anon$4.work(System.scala:63)
	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:226)
	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:226)
	at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
	at sbt.Execute.work(Execute.scala:235)
	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:226)
	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:226)
	at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
	at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
[error] (test:loadedTestFrameworks) org.scalajs.jsenv.ExternalJSEnv$NonZeroExitException: PhantomJS exited with code 2
[error] Total time: 6 s, completed Mar 6, 2016 11:02:16 AM
```

With debug enabled:

```
[debug] > test
[debug] Evaluating tasks: test:test
[debug] Running task... Cancel: Null, check cycles: false, forcegc: true
[debug]
[debug] Initial source changes:
[debug] 	removed:Set()
[debug] 	added: Set()
[debug] 	modified: Set()
[debug] Invalidated products: Set()
[debug] External API changes: API Changes: Set()
[debug] Modified binary dependencies: Set()
[debug] Initial directly invalidated sources: Set()
[debug]
[debug] Sources indirectly invalidated by:
[debug] 	product: Set()
[debug] 	binary dep: Set()
[debug] 	external source: Set()
[debug] All initially invalidated sources: Set()
[debug] Copy resource mappings:
[debug]
[debug]
[debug] Initial source changes:
[debug] 	removed:Set()
[debug] 	added: Set()
[debug] 	modified: Set()
[debug] Invalidated products: Set()
[debug] External API changes: API Changes: Set()
[debug] Modified binary dependencies: Set()
[debug] Initial directly invalidated sources: Set()
[debug]
[debug] Sources indirectly invalidated by:
[debug] 	product: Set()
[debug] 	binary dep: Set()
[debug] 	external source: Set()
[debug] All initially invalidated sources: Set()
[debug] Copy resource mappings:
[debug]
[debug] Loading JSEnv with linked file projects/scala-js-example-app/target/scala-2.11/example-test-fastopt.js
[trace] Stack trace suppressed: run last test:loadedTestFrameworks for the full output.
[error] (test:loadedTestFrameworks) org.scalajs.jsenv.ExternalJSEnv$NonZeroExitException: PhantomJS exited with code 2
[error] Total time: 6 s, completed Mar 6, 2016 11:05:14 AM
```

Now this may be a bug with the angular library, however I'd like to see at least some help on what causes phantomjs to exit (such as the javascript console output)
Poking around in the scalajs sbt plugin it seems that exit code 2 is caused by the webpage calling onError:

https://github.com/scala-js/scala-js/blob/f27a42c6aa83833bb44e9efe8d58b426131893f9/js-envs/src/main/scala/org/scalajs/jsenv/phantomjs/PhantomJSEnv.scala#L443

Thanks for the help,

**Comment by axos88** [comment]
@sjrd?

**Comment by axos88** [comment]
sbt run prints out "Syntax Error"

**Comment by axos88** [comment]
```
> myproject-js/run
[debug]
[debug] Initial source changes:
[debug]     removed:Set()
[debug]     added: Set()
[debug]     modified: Set()
[debug] Removed products: Set()
[debug] External API changes: API Changes: Set()
[debug] Modified binary dependencies: Set()
[debug] Initial directly invalidated sources: Set()
[debug]
[debug] Sources indirectly invalidated by:
[debug]     product: Set()
[debug]     binary dep: Set()
[debug]     external source: Set()
[debug] All initially invalidated sources: Set()
[debug] Copy resource mappings:
[debug]
[debug] Loading JSEnv with linked file myproject/scalajs/target/scala-2.11/myproject-js-fastopt.js
[info] Running JSBoot
[debug] with JSEnv ExternalJSEnv for PhantomJS
[debug] PhantomJS using webpage launcher at: /var/folders/r9/y97n1_g56jq76xcgv2d9n8jm000___/T/phantomjs-launcher-webpage2601653425101094794.html
[debug] PhantomJS using launcher at: /var/folders/r9/y97n1_g56jq76xcgv2d9n8jm000___/T/phantomjs-launcher7773494781003381759.js
[debug] Starting process: phantomjs /var/folders/r9/y97n1_g56jq76xcgv2d9n8jm000___/T/phantomjs-launcher7773494781003381759.js
SyntaxError: Parse error
org.scalajs.jsenv.ExternalJSEnv$NonZeroExitException: PhantomJS exited with code 2
    at org.scalajs.jsenv.ExternalJSEnv$AbstractExtRunner.waitForVM(ExternalJSEnv.scala:107)
    at org.scalajs.jsenv.ExternalJSEnv$ExtRunner.run(ExternalJSEnv.scala:156)
    at org.scalajs.sbtplugin.ScalaJSPluginInternal$.org$scalajs$sbtplugin$ScalaJSPluginInternal$$jsRun(ScalaJSPluginInternal.scala:640)
    at org.scalajs.sbtplugin.ScalaJSPluginInternal$$anonfun$57$$anonfun$apply$40$$anonfun$apply$41.apply(ScalaJSPluginInternal.scala:700)
    at org.scalajs.sbtplugin.ScalaJSPluginInternal$$anonfun$57$$anonfun$apply$40$$anonfun$apply$41.apply(ScalaJSPluginInternal.scala:694)
    at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
[trace] Stack trace suppressed: run last myproject-js/compile:run for the full output.
[error] (myproject-js/compile:run) org.scalajs.jsenv.ExternalJSEnv$NonZeroExitException: PhantomJS exited with code 2
[error] Total time: 4 s, completed Jun 4, 2016 6:33:02 PM
```

**Comment by axos88** [comment]
sbt definition is:

```
  lazy val scalajs = Project(id = s"$projectName-js", base = file("scalajs"))
    .enablePlugins(ScalaJSPlugin)
    .settings(
      libraryDependencies ++= Seq(
         "com.lihaoyi"      %%%!  "utest"       % "0.4.3"   % "test"
        ,"org.scala-js"     %%%!  "scalajs-dom" % "0.9.0"
        ,"com.greencatsoft" %%%! "scalajs-angular"   % "0.6"

      )

      ,jsDependencies ++= Seq(
         RuntimeDOM
        ,"org.webjars.bower" % "angular" % "1.5.6" / "angular.js"
      )


      ,scalaJSUseRhino in Global := false

      ,testFrameworks += new TestFramework("utest.runner.Framework")
      ,persistLauncher in Compile := true
      ,persistLauncher in Test := false
      ,emitSourceMaps := true
    )
```

**Comment by axos88** [comment]
Just figured this out. The angular.js contains comments that contain code examples, which contain <script></sciprt tags. Due to how phantomJS concats the JS code, the begining of the included angular <script> tag is matched with one in the comments, causing the parse error.

**Comment by axos88** [comment]
The string could occur as a string literal as well, probably should escape that somehow...

**Comment by axos88** [comment]
Adding 
```
<script type="text/javascript">
//<![CDATA[
...code...
//]]>
</script>
```

Wouldn't solve the problem entirely, but would surely solve most of the occurences.

**Comment by axos88** [comment]
fixed by commit f552597d086a5ac924001e79e2e733355aea07aa via #2377

**Comment by sjrd** [comment]
Hey @axos88. I'm sorry we forgot to follow up on this issue. It must have fallen through the cracks, never to be found again :-(

I'm glad you managed to troubleshoot it down, though! Since you're saying it's fixed by #2377, I'll close this one as duplicate of #2376.
